AsomamecoApi/test/Controllers/RolesControllerTests.cs

```diff
@@ -0,0 +1,265 @@
+using src.Services;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace test.Controllers
+{
+    public class RolesControllerTests
+    {
+        private readonly IFixture _fixture;
+        private readonly Mock<IRoleService> _serviceMock;
+        private readonly RoleController _controller;
+
+        public RolesControllerTests()
+        {
+            _fixture = new Fixture();
+            _serviceMock = _fixture.Freeze<Mock<IRoleService>>();
+            _controller = new RoleController(_serviceMock.Object);
+        }
+
+        // Test Get All
+        [Fact]
+        public async Task GetAll_ShouldReturnOkResponse_WhenDataFound()
+        {
+            // Arrange
+            var rolesMock = new List<Role>()
+            {
+                new Role { Id = 1, Description = "Admin"},
+                new Role { Id = 2,Description = "Member"}
+            };
+
+            _serviceMock.Setup(service => service.GetAll(1, 10)).ReturnsAsync(rolesMock);
+
+            // Act
+            var result = await _controller.GetRoles().ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Should().BeAssignableTo<ActionResult<IEnumerable<Event>>>();
+            result.Result.Should().BeOfType<OkObjectResult>();
+            result.Result.As<OkObjectResult>().Value
+                .Should()
+                .NotBeNull()
+                .And.BeOfType(rolesMock.GetType());
+            _serviceMock.Verify(service => service.GetAll(1, 10), Times.Once);
+        }
+
+        [Fact]
+        public async Task GetAll_ShouldReturnNoContent_WhenNoDataFound()
+        {
+            // Arrange
+            _serviceMock.Setup(service => service.GetAll(1, 10)).ReturnsAsync((List<Role>)null);
+
+            // Act
+            var result = await _controller.GetRoles().ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Result.Should().BeOfType<NoContentResult>();
+            _serviceMock.Verify(service => service.GetAll(1, 10), Times.Once);
+        }
+
+        [Fact]
+        public async Task GetAll_ShouldReturnInternalServerError_WhenExceptionThrown()
+        {
+            // Arrange
+            _serviceMock.Setup(service => service.GetAll(1, 10)).ThrowsAsync(new Exception("Some error"));
+
+            // Act
+            var result = await _controller.GetRoles().ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Result.Should().BeOfType<ObjectResult>();
+            result.Result.As<ObjectResult>().StatusCode.Should().Be(StatusCodes.Status500InternalServerError);
+            _serviceMock.Verify(service => service.GetAll(1, 10), Times.Once);
+        }
+
+        // Test Get By ID
+        [Fact]
+        public async Task GetByID_ShouldReturnOkResponse_WhenValidInput()
```

Comment:
Este test falla: Message: 
  Expected result to be assignable to Microsoft.AspNetCore.Mvc.ActionResult`1[src.Models.Event], but Microsoft.AspNetCore.Mvc.ActionResult`1[src.Models.Role] is not.